### DISCUSSION TITLE
Koord iOS autobuild

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#2105] [Ideas]
https://github.com/koord-live/koord-realtime/blob/a3253edac526fdc5aa475d358a56eea524ffb340/.github/workflows/autobuild.yml#L127

@danryu seemed to at least partly get an iOS build working on his fork. Although it doesn't generate an .ipa file, I assume the work could be continued to provide installable .ipa files for iOS (handy for sideloading). As far as I see, it's based on the macOS build script.

Maybe that's something for @emlynmac and @ngocdh ?

**Comment by ngocdh** [comment]
That's great news. Could we maybe even ask @danryu for a ios build PR?

**Comment by danryu** [comment]
Hi guys

Yes, this is part of the work I did on the fork a little while back, before I became overwhelmed with other project workloads.
However, I am coming back to jamulus dev in a week or so when my workload shifts.

I would be delighted to provide PRs - currently I have a lot of changes blended in one branch, many of which you guys probably don't want.
I assume you'd also want to have a feature-per-PR, rather than bundle a lot of things together.

**Reply by ngocdh** [reply]
Awesome @danryu I look forward to your PRs.
In my limited experience indeed one PR per feature is better.

About iOS build, did you manage to generate.ipa file using GitHub actions?

**Reply by danryu** [reply]
The iOS autobuild is currently just generating the .app file - I don't think I've attempted the ipa target yet.
You can see the last successful build here: https://github.com/koord-live/koord-realtime/runs/3243694115

Feel free to provide me with notes or docs for the full build requirements, they'll come in handy!

**Reply by ngocdh** [reply]
@emlynmac is the one with iOS build and publishing experience here. What do you think @emlynmac? From .app to publishable .ipa, is it something we could make happen?

**Comment by danryu** [comment]
I noticed this CMake macro repo which looks like it will probably be useful:
https://github.com/OlivierLDff/QtIosCMake

I'll be looking at integrating that (or something like it) in a couple of weeks when my current workload dies down.

**Comment by ann0see** [comment]
Ok. Also did some work on generating .xcarchive files. They should be installed to a iDevice by https://github.com/maciekish/iReSign anyone who wants to test, can do so by downloading the file from here: https://github.com/ann0see/jamulus/actions/runs/1807341897

I don't know how much we can streamline the process, but I think it's a good start

**Reply by danryu** [reply]
Thanks @ann0see - my schedule for iOS stuff got pushed back a bit but I will look at this when I flesh out the autobuild (soon).

**Reply by ann0see** [reply]
See #2355 I think I got IPA creation to work
